src/main.c

```diff
@@ -1,51 +1,51 @@
 #include <stdio.h>
 #include <stdint.h>
+
 #include "lcd.h"
 #include "cpu.h"
 #include "rom.h"
 #include "mmu.h"
+#include "errcodes.h"
 
 int main(void) {
-    uint8_t ret = rom_load("./roms/hello-world.gb");
+    ErrorCode ret = rom_load("./roms/hello-world.gb");
     if (ret) {
         fprintf(stderr, "Failed to load ROM\n");
-        exit(-1);
+        exit((int)ret);
     }
     printf("Successfully initialized ROM\n");
 
     ret = mem_init();
     if (ret) {
         fprintf(stderr, "Failed to initialize MMU\n");
-        exit(-2);
+        exit((int)ret);
     }
     printf("Successfully initialized MMU\n");
-    
+
 
     ret = lcd_load();
     if (ret) {
         fprintf(stderr, "Failed to initialize LCD\n");
-        exit(-3);
+        exit((int)ret);
     }
     printf("Successfully initialized LCD\n");
 
 
     cpu_init();
     printf("Successfully initialized CPU\n");
 
-    uint8_t x = 0;
-    while(1) {
-        if (cpu_step()) {
+    uint8_t iterations = 0;
+    while (1) {
+        if (cpu_step())
             break;
-        }
 
-        if (lcd_step()) {
+        if (lcd_step())
             break;
-        }
 
-        x++;
+        iterations++;
```

Comment:
```suggestion
        ++iterations;
```

in order to avoid variable copying, as we only want to increment the variable

Reply:
Considering your first comment for src/main.c:14: This just has to be used in main(). No casting is necessary anywhere else, if ErrorCodes gets used consistently.